### ISSUE TITLE
AIP 4: Fee Split Rewards for Token Holders

### ISSUE THREAD

**Issue by benevolentdictator** [airswap/airswap-aips#4]
Summary

AirSwap has historically struggled to adequately drive demand for the AST token, in spite of the various mechanisms employed to do so. The only solution to this problem, may perhaps be the simplest; reward token holders with a portion of all tokens traded on the AirSwap network. All previous AIPs will increase the number of AST tokens in circulation, and increase sell pressure. There must be a countering force to encourage individuals to both keep and accumulate AST tokens.  

Specification

Like many other projects, Polkadot, Ethereum, etc, AirSwap can provide the facility for token holders to stake their AST tokens and be rewarded, not just with AST tokens, but whatever tokens are traded on the network. This was successfully implemented by Coss (now Cos), and other exchanges. This ensures that the tokens themselves have intrinsic value. 

Rewards can be calculated based on the number of tokens staked, and contributions to voting and general governance. 

Rationale 

This proposal aims to counter the sell pressure that will be created by airdrops and AST rewards given to liquidity providers. It is absolutely necessary to drive demand for AST while ensuring token holder engagement, and by extension, the success of the project. 

Copyright

Copyright and related rights waived via CC0.

**Comment by asiangoats** [comment]
I think this proposal will definitely drive demand for the underlying token (AST). But one of the primary features that makes airswap stand out from the rest is the fee free trading environment. But obviously that hasn't created significant demand for airswap products. Maybe the larger market is telling us that they really don't mind paying fees as long as there's enough liquidity, and liquidity only comes when there's incentives to add liquidity.

**Comment by asiangoats** [comment]
2. Also fees should also be split between LPs as well, assuming that this proposal was implemented. LPs are like the backbone of a DEXs. So we have to provide good incentives to attract more LPs. As well as a solid liquidity mining campaign to on board traders/users.

**Comment by fawazahmed** [comment]
Agreed. An Airdrop would likely hurt Airswap's momentum.

**Comment by pg024** [comment]
> I think this proposal will definitely drive demand for the underlying token (AST). But one of the primary features that makes airswap stand out from the rest is the fee free trading environment. But obviously that hasn't created significant demand for airswap products. Maybe the larger market is telling us that they really don't mind paying fees as long as there's enough liquidity, and liquidity only comes when there's incentives to add liquidity.

Well since it's all erc-20 compatible AST holders can just receive fees in the form of the traded tokens. 
Take REN as example, darknode holders bond 100k REN but get rewards in BTC, BCH, ZEC (currently active tokens) and front users in order to use the platform do not need to touch any REN at all.

This would make of AST a work token without adding to the platform any friction.

**Comment by AirswapWhale** [comment]
I will support this proposal IF token holders are required to stake/lock their tokens to be eligible for the fee rewards.

We should reward active and participating holders while creating value accruing mechanisms.

This way tokens will be removed from circulation reducing supply and holders can be rewarded.

**Comment by benevolentdictator** [comment]
> I will support this proposal IF token holders are required to stake/lock their tokens to be eligible for the fee rewards.
> 
> We should reward active and participating holders while creating value accruing mechanisms.
> 
> This way tokens will be removed from circulation reducing supply and holders can be rewarded.
> 
> This is one of the most important AIPs along with AIP5: Token Burn. It will definitely give token holders an incentive to continue supporting the protocol.

I agree that token holders should be required to stake/lock their tokens in order to be eligible for fee split rewards. The mechanism currently employed by synthetix is workable if a layer 2 solution is implemented to ensure fees for staking and claiming are low. Any fee split rewards not claimed within the prescribed period would be forfeited and rolled over and added to the total fee rewards for the next period

**Comment by Carl-iCoin** [comment]
this is all great stuff.  When can we set this in motion?  There are no downsides to this, only upsides.

**Comment by AirswapWhale** [comment]
I would also like to add the following:

Metaswap already has fees. We'd need to add fees on top of it. That could be a burden to bootstrap volume. So for a certain X amount of time  (For example: until airswap hits $1 billion of total swaps) I suggest the following:

Not every token will be burnt if the Burn AIP passes. Some will be left. So place them in a treasury. Then reach consensus on how much extra fee (for rewards) Airswap will charge. Then during that time, instead of taxing that extra fee to Metaswap users, we take it from the treasury.

This way:
- token holders are still rewarded
- metamask users won't opt out of Airswap 

Then once things are okay and we hit bigger vumes, we introduce that fee. Also, to be elogible for any kind of rewards you would need to stake your AST tokens. This would offset any inflation from the initial rewards (that will be a small amount because it is only for a short time).


Just a suggestion to find a way around the "expensive fees" people are claiming and also still rewarding tooen holders AND removing tokens from circulation (via staking)

**Comment by benevolentdictator** [comment]
> I would also like to add the following:
> 
> Metaswap already has fees. We'd need to add fees on top of it. That could be a burden to bootstrap volume. So for a certain X amount of time (For example: until airswap hits $1 billion of total swaps) I suggest the following:
> 
> Not every token will be burnt if the Burn AIP passes. Some will be left. So place them in a treasury. Then reach consensus on how much extra fee (for rewards) Airswap will charge. Then during that time, instead of taxing that extra fee to Metaswap users, we take it from the treasury.
> 
> This way:
> 
> * token holders are still rewarded
> * metamask users won't opt out of Airswap
> 
> Then once things are okay and we hit bigger vumes, we introduce that fee. Also, to be elogible for any kind of rewards you would need to stake your AST tokens. This would offset any inflation from the initial rewards (that will be a small amount because it is only for a short time).
> 
> Just a suggestion to find a way around the "expensive fees" people are claiming and also still rewarding tooen holders AND removing tokens from circulation (via staking)

Great suggestions, but I think it's important to note that the largest dex, uniswap, also charges fees. That said, no issue whatsoever with implementing as outlined above.

**Comment by AirswapWhale** [comment]
> > I would also like to add the following:
> > Metaswap already has fees. We'd need to add fees on top of it. That could be a burden to bootstrap volume. So for a certain X amount of time (For example: until airswap hits $1 billion of total swaps) I suggest the following:
> > Not every token will be burnt if the Burn AIP passes. Some will be left. So place them in a treasury. Then reach consensus on how much extra fee (for rewards) Airswap will charge. Then during that time, instead of taxing that extra fee to Metaswap users, we take it from the treasury.
> > This way:
> > 
> > * token holders are still rewarded
> > * metamask users won't opt out of Airswap
> > 
> > Then once things are okay and we hit bigger vumes, we introduce that fee. Also, to be elogible for any kind of rewards you would need to stake your AST tokens. This would offset any inflation from the initial rewards (that will be a small amount because it is only for a short time).
> > Just a suggestion to find a way around the "expensive fees" people are claiming and also still rewarding tooen holders AND removing tokens from circulation (via staking)
> 
> Great suggestions, but I think it's important to note that the largest dex, uniswap, also charges fees. That said, no issue whatsoever with implementing as outlined above.

@benevolentdictator , the issue is most people will use Airswal via Metamask. Metamask already charges fees. With thia, we add even more fees. So it's Metamask Fees + Uniswal Fees. If it's above Uniswap fees for example it could detract people from using it.

I want this to be approved but we need to be careful with what we charge otherwise when people use Metamask their orders will be routed through other protocols and not Airswap.

**Comment by benevolentdictator** [comment]
> > > I would also like to add the following:
> > > Metaswap already has fees. We'd need to add fees on top of it. That could be a burden to bootstrap volume. So for a certain X amount of time (For example: until airswap hits $1 billion of total swaps) I suggest the following:
> > > Not every token will be burnt if the Burn AIP passes. Some will be left. So place them in a treasury. Then reach consensus on how much extra fee (for rewards) Airswap will charge. Then during that time, instead of taxing that extra fee to Metaswap users, we take it from the treasury.
> > > This way:
> > > 
> > > * token holders are still rewarded
> > > * metamask users won't opt out of Airswap
> > > 
> > > Then once things are okay and we hit bigger vumes, we introduce that fee. Also, to be elogible for any kind of rewards you would need to stake your AST tokens. This would offset any inflation from the initial rewards (that will be a small amount because it is only for a short time).
> > > Just a suggestion to find a way around the "expensive fees" people are claiming and also still rewarding tooen holders AND removing tokens from circulation (via staking)
> > 
> > 
> > Great suggestions, but I think it's important to note that the largest dex, uniswap, also charges fees. That said, no issue whatsoever with implementing as outlined above.
> 
> @benevolentdictator , the issue is most people will use Airswal via Metamask. Metamask already charges fees. With thia, we add even more fees. So it's Metamask Fees + Uniswal Fees. If it's above Uniswap fees for example it could detract people from using it.
> 
> I want this to be approved but we need to be careful with what we charge otherwise when people use Metamask their orders will be routed through other protocols and not Airswap.

Agreed. Volume is key, and fees should be set with competitors in mind.

**Comment by dmosites** [comment]
This has been consolidated into #7
